src/tests/playwright/pages/util/audit.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Check if |actual| is close to |expected| using the given relative error
+ * |threshold|.
+ * @param {number} actual
+ * @param {number} expected
+ * @param {number} threshold
+ * @return {[boolean, string]} if |actual| is within |threshold| of |expected|
+ */
+export function beCloseTo(actual, expected, threshold) {
+  // The threshold is relative except when |expected| is zero, in which case
+  // it is absolute.
+  const absExpected = expected ? Math.abs(expected) : 1;
+  const error = Math.abs(actual - expected) / absExpected;
+  return [
+    error <= threshold,
+    // eslint-disable-next-line max-len
+    `${ actual } vs ${ expected } | error: ${error} | ${ Math.abs(actual - expected) } diff`,
+  ];
+}
+
+/**
+ * Assigns a given test function to the global window._webAudioTest property.
+ * This function is used to set up a test that can later be executed globally.
+ *
+ * @param {Promise<any>} testPromise - The test function to be assigned to
+ * window._webAudioTest.
+ */
+export const test = (testPromise) => {
+  window._webAudioTest = testPromise;
+  window._webAudioSuite && window._webAudioMutex();
+};
+
+/**
+ * Evaluates a given function by assigning it to window.webAudioEvaluate.
+ * If window._webAudioSuite property is true, the function is assigned directly.
+ * Otherwise, the function is invoked immediately.
+ *
+ * @param {Function} fun - The function to evaluate.
+ * @return {any}
+ */
+export const evaluate = (fun) => window.webAudioEvaluate =
```

Comment:
Regarding 1 & 2, it might be nicer to keep it as `evaluate` because each test HTML page has two parts: a `test` function, and an `evaluate` function. Test is used for timing and evaluate is used to check all the assertions after running.

Reply:
@Kizjkre Acknowledged.

@terryzfeng I understand, but generally the function name with a single verb will cause the name confusion/collision in the future. That's why I am always pushing for the minimum level of specificity when it comes to naming. Can we at least do "evaluate (verb)" + a noun?